<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="BEAST2 (&lt;https://www.beast2.org&gt;) is a widely used
    Bayesian phylogenetic tool, that uses DNA/RNA/protein data
    and many model priors to create a posterior of jointly estimated 
    phylogenies and parameters.
    mcbette allows to do a Bayesian model comparison over some
    site and clock models, 
    using babette (&lt;https://github.com/ropensci/babette/&gt;).">
<title>Model Comparison Using babette • mcbette</title>
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="/apple-touch-icon.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://docs.ropensci.org/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Model Comparison Using babette">
<meta property="og:description" content="BEAST2 (&lt;https://www.beast2.org&gt;) is a widely used
    Bayesian phylogenetic tool, that uses DNA/RNA/protein data
    and many model priors to create a posterior of jointly estimated 
    phylogenies and parameters.
    mcbette allows to do a Bayesian model comparison over some
    site and clock models, 
    using babette (&lt;https://github.com/ropensci/babette/&gt;).">
<meta property="og:image" content="https://docs.ropensci.org/mcbette/logo.png">
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Matomo --><link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.css">
<script src="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.js" data-cfasync="false"></script><script src="https://ropensci.org/scripts/matomo.js"></script><noscript><p><img src="https://ropensci.matomo.cloud/matomo.php?idsite=1&amp;rec=1" style="border:0;" alt=""></p></noscript>
<!-- End Matomo Code -->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    <a href="https://ropensci.org" class="external-link"><img src="https://ropensci.org/img/icon_short_white.svg" id="hexlogo" alt="rOpenSci"></a>
    <a class="navbar-brand me-2" href="index.html">mcbette</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.15.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"></ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/demo.html">Demo</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ropensci/mcbette/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="mcbette">mcbette<a class="anchor" aria-label="anchor" href="#mcbette"></a>
</h1>
</div>
<!-- badges: start -->
<!-- badges: end --><p><img src="reference/figures/mcbette_logo.png"></p>
<p><code>mcbette</code> allows for doing a Model Comparison using <code>babette</code> (hence the name), that is, given a (say, DNA) alignment, it compares multiple phylogenetic inference models to find the model that is likeliest to generate that alignment. With this, one can find the phylogenetic inference model that is simple enough, but not too simple.</p>
<p>To do so, <code>mcbette</code> uses <code>babette</code> [Bilderbeek and Etienne, 2018] with the addition of using the BEAST2 [Bouckaert et al., 2019] nested sampling package as described in [Russell et al., 2019].</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p><span class="emoji" data-emoji="warning">⚠️</span> <code>mcbette</code> only works on Linux and Mac.</p>
<p><code>mcbette</code> depends on the <a href="https://cran.r-project.org/package=rJava" class="external-link">rJava</a> and <a href="https://cran.r-project.org/package=Rmpfr" class="external-link">Rmpfr</a> packages.</p>
<p>On Linux, to install these, do (as root):</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>apt install r<span class="sc">-</span>cran<span class="sc">-</span>rjava libmpfr<span class="sc">-</span>dev</span></code></pre></div>
<p>After this, installing <code>mcbette</code> is rather easy:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"mcbette"</span><span class="op">)</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"richelbilderbeek/beastierinstall"</span><span class="op">)</span></span>
<span><span class="fu">beastierinstall</span><span class="fu">::</span><span class="fu">install_beast2</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"richelbilderbeek/mauricerinstall"</span><span class="op">)</span></span>
<span><span class="fu">mauricerinstall</span><span class="fu">::</span><span class="fu">install_beast2_pkg</span><span class="op">(</span><span class="st">"NS"</span><span class="op">)</span></span></code></pre></div>
<p>Note that <code>mcbette</code> uses the non-CRAN extensions <a href="https://github.com/richelbilderbeek/beastierinstall" class="external-link">beastierinstall</a> and <a href="https://github.com/richelbilderbeek/mauricerinstall" class="external-link">mauricerinstall</a> to install and uninstall BEAST2 and its packages.</p>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/mcbette">mcbette</a></span><span class="op">)</span></span></code></pre></div>
<p>Suppose we have a DNA alignment, obtained by sequencing multiple species:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fasta_filename</span> <span class="op">&lt;-</span> <span class="fu">beautier</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/beautier/reference/get_beautier_path.html" class="external-link">get_beautier_path</a></span><span class="op">(</span><span class="st">"anthus_aco_sub.fas"</span><span class="op">)</span></span>
<span><span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/image.DNAbin.html" class="external-link">image.DNAbin</a></span><span class="op">(</span><span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.dna.html" class="external-link">read.FASTA</a></span><span class="op">(</span><span class="va">fasta_filename</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-example_alignment-1.png" width="100%"></p>
<p>Note that this alignment holds too little information to really base a publishable research on.</p>
<p>To create a posterior distribution of phylogenies from this alignment, one needs to specify an inference model. An inference model is (among others) a combination of a site model, clock model and tree model. See the ‘Available models’ section to see the available models.</p>
<p>In this example we let two inference models compete.</p>
<p>Here is the default inference model:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">inference_model_1</span> <span class="op">&lt;-</span> <span class="fu">beautier</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/beautier/reference/create_ns_inference_model.html" class="external-link">create_ns_inference_model</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span></span>
<span>    <span class="va">inference_model_1</span><span class="op">$</span><span class="va">site_model</span><span class="op">$</span><span class="va">name</span>,</span>
<span>    <span class="va">inference_model_1</span><span class="op">$</span><span class="va">clock_model</span><span class="op">$</span><span class="va">name</span>,</span>
<span>    <span class="va">inference_model_1</span><span class="op">$</span><span class="va">tree_prior</span><span class="op">$</span><span class="va">name</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; JC69 strict yule</span></span></code></pre></div>
<p>The JC69 site model assumes that the four DNA nucleotides are equally likely to mutate from/to. The strict clock model assumes that mutation rates of all species are equal. The Yule tree model assumes that new species form at a constant rate for an extinction rate of zero.</p>
<p>The competing model has a different site, clock and tree model:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">inference_model_2</span> <span class="op">&lt;-</span> <span class="va">inference_model_1</span></span>
<span><span class="va">inference_model_2</span><span class="op">$</span><span class="va">site_model</span> <span class="op">&lt;-</span> <span class="fu">beautier</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/beautier/reference/create_hky_site_model.html" class="external-link">create_hky_site_model</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">inference_model_2</span><span class="op">$</span><span class="va">clock_model</span> <span class="op">&lt;-</span> <span class="fu">beautier</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/beautier/reference/create_rln_clock_model.html" class="external-link">create_rln_clock_model</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">inference_model_2</span><span class="op">$</span><span class="va">tree_prior</span> <span class="op">&lt;-</span> <span class="fu">beautier</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/beautier/reference/create_bd_tree_prior.html" class="external-link">create_bd_tree_prior</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>The HKY site model assumes that DNA substitution rates differ between transitions (purine-to-purine or pyrimidine-to-pyrimidine) and translations (purine-to-pyrimidine or the other way around). The relaxed log-normal clock model assumes that mutation rates of all species are differ, where all these rates together follow a log-normal distribution. The birth-death tree model assumes that new species form and go extinct at a constant rate.</p>
<p><code>mcbette</code> shows the evidence (also called marginal likelihood) for each inference model, which is the likelihood that a model has generated the data.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="reference/can_run_mcbette.html">can_run_mcbette</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">marg_liks</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/est_marg_liks.html">est_marg_liks</a></span><span class="op">(</span></span>
<span>    fasta_filename <span class="op">=</span> <span class="va">fasta_filename</span>,</span>
<span>    inference_models <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">inference_model_1</span>, <span class="va">inference_model_2</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>Here we display the marginal likelihoods as a table:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="reference/can_run_mcbette.html">can_run_mcbette</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">marg_liks</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>The most important result are the model weights. When a model’s weight is very close to one, one would prefer to use that inference model in doing a Bayesian inference. If these model weights are rather similar, one could argue to use either model.</p>
<p>Here we display the marginal likelihoods as a barplot:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="reference/can_run_mcbette.html">can_run_mcbette</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="reference/plot_marg_liks.html">plot_marg_liks</a></span><span class="op">(</span><span class="va">marg_liks</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="available-models">Available models<a class="anchor" aria-label="anchor" href="#available-models"></a>
</h2>
<p>The available site models:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">beautier</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/beautier/reference/get_site_model_names.html" class="external-link">get_site_model_names</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "JC69" "HKY"  "TN93" "GTR"</span></span></code></pre></div>
<p>The available clock models:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">beautier</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/beautier/reference/get_clock_model_names.html" class="external-link">get_clock_model_names</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "relaxed_log_normal" "strict"</span></span></code></pre></div>
<p>The available tree models:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">beautier</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/beautier/reference/get_tree_prior_names.html" class="external-link">get_tree_prior_names</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "birth_death"                    "coalescent_bayesian_skyline"   </span></span>
<span><span class="co">#&gt; [3] "coalescent_constant_population" "coalescent_exp_population"     </span></span>
<span><span class="co">#&gt; [5] "yule"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="documentation">Documentation<a class="anchor" aria-label="anchor" href="#documentation"></a>
</h2>
<ul>
<li>The mcbette vignette</li>
<li><a href="https://ropensci.org/blog/2020/01/28/babette/" class="external-link">rOpenSci blog post: Call BEAST2 for Bayesian evolutionary analysis from R</a></li>
<li><a href="https://ropensci.org/blog/2020/12/01/mcbette-selecting-the-best-inference-model/" class="external-link">rOpenSci blog post: Selecting the Best Phylogenetic Evolutionary Model</a></li>
</ul>
</div>
<div class="section level2">
<h2 id="faq">FAQ<a class="anchor" aria-label="anchor" href="#faq"></a>
</h2>
<div class="section level3">
<h3 id="under-which-platforms-does-mcbette-work">Under which platforms does <code>mcbette</code> work?<a class="anchor" aria-label="anchor" href="#under-which-platforms-does-mcbette-work"></a>
</h3>
<p><code>mcbette</code> only works on Linux and Mac, because BEAST2 package management only works on those platforms.</p>
</div>
<div class="section level3">
<h3 id="how-do-i-let-mcbette-compare-all-models">How do I let mcbette compare all models?<a class="anchor" aria-label="anchor" href="#how-do-i-let-mcbette-compare-all-models"></a>
</h3>
<p>First, this is impossible, as there are infinitely many inference models possible.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">inference_models</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">i</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">site_model</span> <span class="kw">in</span> <span class="fu">beautier</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/beautier/reference/create_site_models.html" class="external-link">create_site_models</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw">for</span> <span class="op">(</span><span class="va">clock_model</span> <span class="kw">in</span> <span class="fu">beautier</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/beautier/reference/create_clock_models.html" class="external-link">create_clock_models</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw">for</span> <span class="op">(</span><span class="va">tree_prior</span> <span class="kw">in</span> <span class="fu">beautier</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/beautier/reference/create_tree_priors.html" class="external-link">create_tree_priors</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="va">inference_models</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">beautier</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/beautier/reference/create_ns_inference_model.html" class="external-link">create_ns_inference_model</a></span><span class="op">(</span></span>
<span>        site_model <span class="op">=</span> <span class="va">site_model</span>,</span>
<span>        clock_model <span class="op">=</span> <span class="va">clock_model</span>,</span>
<span>        tree_prior <span class="op">=</span> <span class="va">tree_prior</span></span>
<span>      <span class="op">)</span></span>
<span>      <span class="va">i</span> <span class="op">&lt;-</span> <span class="va">i</span> <span class="op">+</span> <span class="fl">1</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="op">}</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>Now, <code>inference_models</code> holds a list of inference models, to be used with <code><a href="reference/est_marg_liks.html">mcbette::est_marg_liks</a></code>.</p>
</div>
<div class="section level3">
<h3 id="error-direxistsexamples_folder-is-not-true">
<code>Error: dir.exists(examples_folder) is not TRUE</code><a class="anchor" aria-label="anchor" href="#error-direxistsexamples_folder-is-not-true"></a>
</h3>
<p>Currently, this line gives a suboptimal error message:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>beastier<span class="sc">::</span><span class="fu">get_beast2_example_filename</span>(<span class="st">"Primates.nex"</span>)</span></code></pre></div>
<p>The error message is:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>Error<span class="sc">:</span> <span class="fu">dir.exists</span>(examples_folder) is not <span class="cn">TRUE</span></span></code></pre></div>
<p>The error message it should display is:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>Error<span class="sc">:</span> BEAST2 examples folder not found at path <span class="st">'[path]'</span>.</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>Maybe BEAST2 is not installed? </span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>Tip<span class="sc">:</span> run <span class="st">'beastier::install_beast2()'</span></span></code></pre></div>
<p>This will be added in a future version of <code>beastier</code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="using-an-existing-beast2-installation">Using an existing BEAST2 installation<a class="anchor" aria-label="anchor" href="#using-an-existing-beast2-installation"></a>
</h2>
<p>When BEAST2 is already installed, yet at a non-default location, one can use the <code>beast2_bin_path</code> argument in <code>create_mcbette_beast2_options</code>.</p>
</div>
<div class="section level2">
<h2 id="code-of-conduct">Code of conduct<a class="anchor" aria-label="anchor" href="#code-of-conduct"></a>
</h2>
<p>Please note that this package is released with a <a href="https://ropensci.org/code-of-conduct/" class="external-link">Contributor Code of Conduct</a>. By contributing to this project, you agree to abide by its terms.</p>
</div>
<div class="section level2">
<h2 id="links">Links<a class="anchor" aria-label="anchor" href="#links"></a>
</h2>
<ul><li><a href="https://CRAN.R-project.org/package=mcbette" class="external-link">‘mcbette’ CRAN page</a></li></ul>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li>Bilderbeek, Richel JC, and Rampal S. Etienne. “babette: BEAUti 2, BEAST 2 and Tracer for R.” Methods in Ecology and Evolution (2018). <a href="https://doi.org/10.1111/2041-210X.13032" class="external-link uri">https://doi.org/10.1111/2041-210X.13032</a>
</li>
<li>Bouckaert R., Vaughan T.G., Barido-Sottani J., Duchêne S., Fourment M., Gavryushkina A., et al. (2019) BEAST 2.5: An advanced software platform for Bayesian evolutionary analysis. PLoS computational biology, 15(4), e1006650.</li>
<li>Russel, Patricio Maturana, et al. “Model selection and parameter inference in phylogenetics using nested sampling.” Systematic biology 68.2 (2019): 219-233.</li>
</ul>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">beastier</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/beastier/reference/remove_beaustier_folders.html" class="external-link">remove_beaustier_folders</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu">beastier</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/beastier/reference/check_empty_beaustier_folders.html" class="external-link">check_empty_beaustier_folders</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=mcbette" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/ropensci/mcbette/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/ropensci/mcbette/issues" class="external-link">Report a bug</a></li>
</ul>
</div>



<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CONTRIBUTING.html">Contributing guide</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing mcbette</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Richèl J.C. Bilderbeek <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-1107-7049" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>

<div class="r-universe">
<h2 data-toc-skip>R-universe</h2>
<ul class="list-unstyled">
<li><p><a href="https://ropensci.r-universe.dev/mcbette" class="external-link"><img src="https://ropensci.r-universe.dev/badges/mcbette" alt="mcbette status badge"></a></p></li>
</ul>
</div>

<div class="software-peer-review">
<h2 data-toc-skip>Software Peer-Review</h2>
<ul class="list-unstyled">
<li><p><a href="https://github.com/ropensci/software-review/issues/360" class="external-link"><img src="https://badges.ropensci.org/360_status.svg" alt="rOpenSci peer-review"></a></p></li>
</ul>
</div>

  </aside>
</div>


    <footer><!-- begin footer --><div class="footer">
    <div class="container">
        <div class="row start top-4 bottom-8">
            <div class="col-2"> <img id="footerlogo" src="https://ropensci.org/img/icon_short_white.svg">
</div>
            <div class="col-10">
                <div class="row">
                    <div class="col-md-4 col-xs-6">
                        <a href="https://github.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-github"></div></a>
                        <a href="https://github.com/ropenscilabs" target="_blank" class="external-link"><div class="icon fa fa-flask"></div></a>
                        <a href="https://hachyderm.io/@ropensci" target="_blank" class="external-link"><div class="icon fab fa-mastodon"></div></a>
                        <a href="https://vimeo.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-vimeo"></div></a>
                    </div>
                </div>
                <div class="row top-4">
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">About</h5>
                            <li><a href="https://ropensci.org/about" class="external-link">About rOpenSci</a></li>
                            <li><a href="https://ropensci.org/software-review" class="external-link">Software Review</a></li>
                            <li><a href="https://ropensci.org/about#team" class="external-link">Our Team</a></li>
                            <li><a href="https://ropensci.org/careers" class="external-link">Jobs</a></li>
                            <li><a href="https://ropensci.org/donate" class="external-link">Donate</a></li>
                            <li><a href="https://ropensci.org/contact" class="external-link">Contact Us</a></li>
                        </ul>
</div>
                    <div class="col-md-3 col-sm-4">
                        <ul>
<h5 class="bottom-2">Community</h5>
                            <li><a href="https://ropensci.org/community/" class="external-link">Our Community</a></li>
                            <li><a href="https://ropensci.org/commcalls/" class="external-link">Community calls</a></li>
                            <li><a href="https://ropensci.org/events/" class="external-link">Events</a></li>
                            <li><a href="https://discuss.ropensci.org/" class="external-link">Join the Discussion</a></li>
                            <li><a href="https://ropensci.org/code-of-conduct" class="external-link">Code of conduct</a></li>
                        </ul>
</div>
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">Resources</h5>
                            <li><a href="https://ropensci.org/packages/" class="external-link">Packages</a></li>
                            <li><a href="https://ropensci.org/usecases/" class="external-link">Use Cases</a></li>
                            <li><a href="https://ropensci.org/talks-papers/" class="external-link">Talks &amp; Publications</a></li>
                            <li><a href="https://docs.ropensci.org/" class="external-link">Documentation</a></li>
                            <li><a href="https://ropensci.org/news/" class="external-link">Newsletter</a></li>
                            <li><a href="https://ropensci.org/how-to-cite-ropensci/" class="external-link">Cite rOpenSci</a></li>
                        </ul>
</div>
                    <div class="col-md-4 col-xs-12">
                        <h5 class="bottom-2"></h5>

                        <p>rOpenSci is a fiscally sponsored project of <a href="http://numfocus.org" class="external-link">NumFOCUS</a>.</p>

                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
<!-- / end footer -->


    </footer>
</div>

  

  

  </body>
</html>
